### ISSUE TITLE
mpv framerate tanks after a certain resolution

### ISSUE THREAD

**Issue by hegdenischay** [mpv-player/mpv#13273]
# Summary

mpv drops frames dramatically after it reaches a certain resolution. Very similar to #8613, but it only happens with `scale=ewa_lanczossharp`. `cscale=ewa_lanczossharp` and `video-sync=display-resample` work without issues on my machine.

A screenshot of [this](https://www.youtube.com/watch?v=uLnMg0GNzNQ) video to show my point:
![Screenshot_20240113_144022-3](https://github.com/mpv-player/mpv/assets/28584388/7044b48d-7924-478c-b326-b9a303100450)

Corresponding `mpv.conf`:

```
hwdec=vaapi-copy
#profile=gpu-hq
scale=ewa_lanczossharp
hdr-peak-percentile=99.995
hdr-contrast-recovery=0.30
deband=yes
```

### Important Information
- mpv version: v0.37.0
- Linux distro: Arch Linux
- Source of mpv: distro-supplied
- GPU: Intel HD Graphics 4400 (Intel i5-4300U CPU)

### Reproduction steps

Use `scale=ewa_lanczossharp` in `mpv.conf`, or have `profile=gpu-hq`

### Expected behavior

No frame rate drops for a static image

### Actual behavior

Lots of dropped frames

### Log file
[mpv_log.txt](https://github.com/mpv-player/mpv/files/13927546/mpv_log.txt)

### Sample files
https://0x0.st/Hlfc.mp4

**Comment by llyyr** [comment]
An HD 4400 is probably too weak to both upscale and deband, using either faster scaler or not enabling debanding should fix this. Debanding that video is useless anyway, and it'll be removed from the high-quality profile soon. #13172

**Comment by hegdenischay** [comment]
I agree that this video isn't mjpeg or something, but I don't think this has anything to do with debanding.
I turned debanding off and tested it again with mpv v0.37.0 and mpv v0.36.0 (compiled from the source code given in GitHub releases).
Whatever it is, it appears to be a recent change.

The same `mpv.conf` was used:
```
hwdec=vaapi-copy
#profile=gpu-hq
scale=ewa_lanczossharp
#cscale=ewa_lanczossharp
hdr-peak-percentile=99.995
hdr-contrast-recovery=0.30
#deband=yes
```
## mpv v0.37.0:
![Screenshot_20240114_093436](https://github.com/mpv-player/mpv/assets/28584388/4797658e-c048-4d4d-b0aa-0fba08ad9711)

Log: 
[mpv_new_log.txt](https://github.com/mpv-player/mpv/files/13930303/mpv_new_log.txt)

## mpv v0.36.0:
![Screenshot_20240114_093536](https://github.com/mpv-player/mpv/assets/28584388/2e12eb1f-67e3-40ea-a2ba-0dfc5bc30781)

Log:
[mpv_old_log.txt](https://github.com/mpv-player/mpv/files/13930306/mpv_old_log.txt)

**Comment by llyyr** [comment]
Try vo=gpu-next? I can't think of what might cause a performance regression on vo_gpu.

You can also try setting `scale=ewa_lanczossharp` and `cscale=linear` on 0.37.0, because after 0.37.0 cscale will automatically follow scale if not specified.

**Comment by hegdenischay** [comment]
Thanks for the quick responses, but there's no obvious improvement. Oh, and since the other issue mentioned the DE, this is happening on KDE plasma as well as a TWM setup that I have where I use dwm with picom as a compositor.

With `vo=gpu`:
![Screenshot_20240114_153836](https://github.com/mpv-player/mpv/assets/28584388/76d0ba90-90b6-48c0-a2c7-96a93aa84966)

Log: 
[mpv_log.txt](https://github.com/mpv-player/mpv/files/13931065/mpv_log.txt)

With `vo=gpu-next`:
![Screenshot_20240114_153915](https://github.com/mpv-player/mpv/assets/28584388/9a7a55fc-c01e-4fe8-8b5d-f0d28a27bcfe)
Log:
[mpv_log_gpu_next.txt](https://github.com/mpv-player/mpv/files/13931066/mpv_log_gpu_next.txt)

**Comment by hegdenischay** [comment]
Found the exact commit that caused my current issue: https://github.com/mpv-player/mpv/commit/44cf6288c73b71d22a98870cb08b1daf9da5e048
Additionally, `scaler-lut-size=8` in `profile=gpu-hq` was committed around 4 months ago, and that somehow escaped my radar.

With `scaler-lut-size=8` on mpv v0.36.0:
![Screenshot_20240116_053519-1](https://github.com/mpv-player/mpv/assets/28584388/fb1dfe6d-8251-4b2d-974a-378b85d73e32)

And with `scaler-lut-size=6` on mpv v0.36.0:
![Screenshot_20240116_053645](https://github.com/mpv-player/mpv/assets/28584388/378f440a-00d5-498a-bd64-38a46d80f777)

I think it'd be a good idea to bring that back as an option, and I have created a PR for this.

**Comment by fritzrehde** [comment]
I'm not sure if this is related, but here is my current experience with a recent mpv version: When increasing the speed of a video, the CPU usage scales nearly proportionally. Is that expected? I have used mpv for a couple of years, and this has never been the case iirc. So currently, playing a video at normal speed uses ~50% CPU, whereas playing at 3x speed causes 120% CPU usage on my laptop (Thinkpad T470P with integrated Intel graphics on Linux).

**Comment by hegdenischay** [comment]
I don't believe this is related. What codec is the video?

**Comment by giorgiopagnoni** [comment]
Might be unrelated but I downgraded from 0.37 to 0.36 on OSx for the same reason.

**Comment by llyyr** [comment]
> Might be unrelated but I downgraded from 0.37 to 0.36 on OSx for the same reason.

`--profile=fast`
